include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -2037,13 +2038,21 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::compute_evaluate_fast(
           if constexpr (is_linear)
             {
               if constexpr (n_components == 1)
-                result =
-                  internal::evaluate_tensor_product_value_and_gradient_linear<
-                    dim,
-                    scalar_value_type,
-                    VectorizedArrayType,
-                    1,
-                    stride_view>(solution_values.data(), unit_point_ptr[qb]);
+                {
+                  if (fe->n_components() == 1)
```

Comment:
I am not happy about this additional runtime check but it is necessary to fix the failing test. Any ideas how to avoid this? @peterrum @kronbichler

Reply:
Due to renumbering?

Reply:
Correct, renumbering is the problem.

Reply:
The option is to fit this into the `linear` template argument that you resolve once in the beginning: We only go into the fast path with linear shape functions in case `fe->n_components() == 1` and must go to another function elsewhere. I am not sure if the performance hit to use the generic `evaluate_tensor_product_value_and_gradient_shapes<...>(...)` is acceptable for the case that we have multiple FE components but only pick one, probably one cannot expect optimal performance in that case anyway because we carry around too many unknowns in the local vectors in that case.

Reply:
Set islinear to false during setup?

Reply:
Thanks for the MPI parallel hint @peterrum @kronbichler ! 🧠 🧠